### DISCUSSION TITLE
Share machines between different computers

### DISCUSSION THREAD

**Discussion by raphaelbadia** [statelyai/xstate#1287] [General]
Hello,

I'm totally new to xstate and I would like to know if what I plan to do is doable.
I'll have a server, its job is to compute, convert and resize pictures. It processes files from a queue.
I would like to implement the server using xcode because it's critical and I don't want it to fail.

I would also like to develop a frontend (react) ui that will be able to issue commands to the remote server. 
I would like to know how I can send events and stay synced with the server state in my UI.
Is it possible ? Any suggestion  ?

I think maybe my case is similar to https://github.com/davidkpiano/xstate/discussions/1255 but I'm not sure...

Thanks !

**Comment by CodingDive** [comment]
Do you want to run the same machine on server and client? Usually servers concern are quite different from UI concerns so my first thought would be to create 2 distinct machines.

**Comment by raphaelbadia** [comment]
I don’t need it to be the same machine, but I’d like to be able to get the server’s machine state from my ui client to report what is happening server side... like converting to mp4... downsizing to 1920x1080... creating thumbail...

**Reply by CodingDive** [reply]
I see. The best thing you can do is to establish a web socket connection between Client and Server and use message passing. You can listen in on the socket connection within your machines by invoking a callback as a service.

**Reply by CodingDive** [reply]
You may end up duplicating some state nodes between your Server and Client machines but since completely different things happen in the state transitions, it's not a big deal.

E.g on the server side you do the heavy image processing and compression within your `actions` while on the client, you may just want to update some loading state or advance a progress bar.
